charms/kserve-controller/tests/integration/sklearn-iris.yaml

```diff
@@ -1,12 +1,14 @@
 apiVersion: "serving.kserve.io/v1beta1"
 kind: "InferenceService"
 metadata:
-  name: "sklearn-iris"
+  name: "sklearn-v2-iris"
 spec:
   predictor:
     model:
       modelFormat:
         name: sklearn
+      protocolVersion: v2
+      runtime: kserve-sklearnserver
```

Comment:
I see KServe docs mentioning this change [here](https://kserve.github.io/website/0.11/modelserving/v1beta1/sklearn/v2/#deploy-the-model-with-rest-endpoint-through-inferenceservice), and I guess since we are using the latest version of KServe we should do it, but they don't really provide a good reason other than:

> Lastly, you will use KServe to deploy the trained model onto Kubernetes. For this, you will just need to use version v1beta1 of the InferenceService CRD and set the protocolVersion field to v2.

@ca-scribner did you see an issue on the CI integration tests?

Reply:
v1/v2 refers to the API the server presents.  v2 adopted a standard data protocol like @dnplas linked.  

Both work, but the current examples show how to use v2 so it seemed better to use here.  We could also keep to v1 - im not sure if they're deprecating v1 any time soon

Reply:
Regarding the question on the `runtime`, KServe has `ClusterServingRuntime` objects that define which image and args to use depending on the runtime that is defined. You can find the out of the box list in https://github.com/kserve/kserve/blob/master/install/v0.11.1/kserve-runtimes.yaml

As the docs say for v2 we need to explicitly define the runtime, thus the change above